servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -400,27 +429,30 @@ private TempBlockMeta createBlockMetaNoLock(long userId, long blockId,
 
   /** This method must be guarded by WRITE lock of mEvictionLock */
   private void freeSpaceInternal(long userId, long availableBytes, BlockStoreLocation location)
-      throws IOException {
+      throws OutOfSpaceException {
     EvictionPlan plan = mEvictor.freeSpaceWithView(availableBytes, location, getUpdatedView());
     // Absent plan means failed to evict enough space.
     if (null == plan) {
-      throw new IOException("Failed to free space: no eviction plan by evictor");
+      throw new OutOfSpaceException("Failed to free space: no eviction plan by evictor");
     }
-
     // 1. remove blocks to make room.
     for (long blockId : plan.toEvict()) {
-      long lockId = mLockManager.lockBlock(userId, blockId, BlockLockType.WRITE);
       try {
-        removeBlockNoLock(userId, blockId);
-        synchronized (mBlockStoreEventListeners) {
-          for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-            listener.onRemoveBlockByWorker(userId, blockId);
+        long lockId = mLockManager.lockBlock(userId, blockId, BlockLockType.WRITE);
+        try {
+          removeBlockNoLock(userId, blockId);
+          synchronized (mBlockStoreEventListeners) {
+            for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+              listener.onRemoveBlockByWorker(userId, blockId);
+            }
           }
+        } catch (IOException ioe) {
+          throw new OutOfSpaceException("Failed to free space: cannot evict block " + blockId, ioe);
```

Comment:
It is not entirely correct here: 
`removeBlockNoLock` can also return `IOException` when the file deletion failed on file system.
In this case, I feel we shall not convert and throw `OutOfSpaceException` anymore, but propagate `IOException`